x-pack/plugins/infra/public/pages/metrics/hosts/hooks/use_host_count.ts

```diff
@@ -79,7 +80,23 @@ export const useHostCount = () => {
       searchCriteria.dateRange.from,
       searchCriteria.dateRange.to,
     ]),
-    parseResponses: normalizeDataSearchResponse,
+    parseResponses: useMemo(
```

Comment:
is this `useMemo` really needed in this context? This is already wrapped inside a hook, isn't it?

Reply:
It is. `parseResponses` is in the `useCallback` dependency array of the`useDataSearch` hook. Since `searchCriteria` keeps changing, it will trigger unnecessary `getRequest` calls. `parseResponses` must only be called when `data.search` has finished fetching the data.